backtracking/minimax.cpp

```diff
@@ -1,27 +1,40 @@
+/**
+ * @file
+ * @brief returns which is the longest/shortest number
+ */
+#include <algorithm>
 #include <cmath>
 #include <iostream>
 #include <vector>
 
-using std::cout;
-using std::endl;
-using std::max;
-using std::min;
-using std::vector;
-
-int minimax(int depth, int node_index, bool is_max, vector<int> scores,
-            int height) {
-    if (depth == height)
+/**
+ * Check which number is the maximum/minimum in the array
+ * @param depth depth of array
+ * @param node_index current index in array
+ * @param is_max if current index is the longest number
+ * @param scores saved numbers in vector
+ * @return if maximum or not
```

Comment:
I don't think the function returns a `true` or `false` value.

Reply:
It returns `is_max` which is a boolean.
```c++
return is_max ? std::max(v1, v2) : std::min(v1, v2);
```

Reply:
> It returns `is_max` which is a boolean.
> ```c++
> return is_max ? std::max(v1, v2) : std::min(v1, v2);
> ```

No, it uses the boolean variable to rework the result of std::max or std::min 
That is an inline if else statement

Reply:
this inline if-else block expands to:
```cpp
if(is_max)
    return std::max(v1, v2);
else
    return std::min(v1, v2);
```